tests/hooks/useTabs.test.js

```diff
@@ -0,0 +1,110 @@
+    test('starts at 0', () => {
+      const { result } = renderHook(() => useTabs())
+      const { activeTab } = result.current
+      expect(activeTab).toBe(0)
+    })
+
+    test('or with value in config', () => {
+      const config = { initialTab: 5 } 
+      const { result } = renderHook(() => useTabs(config))
+      const { activeTab } = result.current
+      expect(activeTab).toBe(5)
+    })
+
+    test('changing with selectTab', () => {
+      const { result } = renderHook(() => useTabs())
+      const { selectTab } = result.current
+      act(() => selectTab(1))
+      const { activeTab } = result.current
+      expect(activeTab).toBe(1)
+    })
+
+    test('not allowing to use disabled', () => {
+      const config = {initialDisabledTabs: [1]}
+      const { result } = renderHook(() => useTabs(config))
+      const { selectTab } = result.current
+      act(() => selectTab(1))
+      const { activeTab } = result.current
+      expect(activeTab).toBe(0)
+    })
+  })
+
+  describe('tracks current disabled tabs', () => {
+    test('starting at []', () => {
+      const { result } = renderHook(() => useTabs())
+      const { disabledTabs } = result.current
+      expect(disabledTabs).toEqual([])
+    })
+
+    test('or with value in config', () => {
+      const config = { initialDisabledTabs: [3, 4, 5] } 
+      const { result } = renderHook(() => useTabs(config))
+      const { disabledTabs } = result.current
+      expect(disabledTabs).toEqual([3, 4, 5])
+    })
+
+    test('changing with disableTab', () => {
+      const { result } = renderHook(() => useTabs())
+      const { disableTab } = result.current
+      act(() => disableTab(1))
+      const { disabledTabs } = result.current
+      expect(disabledTabs).toEqual([1])
+    })
+
+    test('allowing enableTab', () => {
+      const config = {initialDisabledTabs: [1]}
+      const { result } = renderHook(() => useTabs(config))
+      const { enableTab } = result.current
+      act(() => enableTab(1))
+      const { disabledTabs } = result.current
+      expect(disabledTabs).toEqual([])
+    })
+
+    test('when nescessary', () => {
+      const config = {initialDisabledTabs: [2]}
+      const { result } = renderHook(() => useTabs(config))
+      const { enableTab } = result.current
+      act(() => enableTab(1))
+      const { disabledTabs } = result.current
+      expect(disabledTabs).toEqual([2])
+    })
+
+    test('not allowing to disable active', () => {
+      const { result } = renderHook(() => useTabs())
+      const { disableTab } = result.current
+      act(() => disableTab(0))
+      const { disabledTabs } = result.current
+      expect(disabledTabs).toEqual([])
+    })
+
+    test('offering an isEnabled function', () => {
+      const config = {initialDisabledTabs: [1]}
+      const { result } = renderHook(() => useTabs(config))
+      const { isEnabled } = result.current
+      expect(isEnabled(0)).toBe(true)
+      expect(isEnabled(1)).toBe(false)
+    })
+  })
+
+})
```

Comment:
Be clear about which feature you are testing and the expected result:
```javascript
describe('useTabs()', () => {
  describe('activeTab integer response', () => {
    test('returns 0 if not receive an initial value', () => {
      const { result } = renderHook(() => useTabs())
      const { activeTab } = result.current
      expect(activeTab).toBe(0)
    })
  })

  describe('when selectTab function response is called', () => {
    test('activeTab is equal to the selectTab passed parameter', () => {
      const { result } = renderHook(() => useTabs())
      const { selectTab } = result.current
      const value = 1
      act(() => selectTab(value))
      const { activeTab } = result.current
      expect(activeTab).toBe(value)
    })
  })
})
```

Reply:
Refactored the descriptions for more readability.